### PR TITLE
Cem/single molecule

### PR DESCRIPTION
**Pull request by chris-erisyon**
Single molecule report code and notebooks

### CODE REVIEW

File: plaster/run/plots/single_molecule/single_molecule_report.py [hunk 1]
```diff
@@ -0,0 +1,813 @@
...
+    get_index_path,
+)
+
+# Paths and names
+
+# In the functions below:
+#   - 'base_path' refers to the base path for the single_molecule_report directory
+#   - 'run_path' refers to the directory for the run
+
+JOBS_FOLDER = "/erisyon/internal/jobs_folder"
```
Comment (zack-erisyon) [inline plaster/run/plots/single_molecule/single_molecule_report.py:34]: It definitely doesn't feel right do have any paths coded into these modules.
This and all of the following should go through interfaces.

File: plaster/run/plots/single_molecule/single_molecule_report.py [hunk 2]
```diff
@@ -0,0 +1,813 @@
...
+# ------------------------------------------------------------
+# Helpers
+# ------------------------------------------------------------
+
+# ------------------------------
+# Config/Index Files
+# ------------------------------
+
+
+def get_run_uuid(run_path):
```
Comment (zack-erisyon) [inline plaster/run/plots/single_molecule/single_molecule_report.py:70]: This and the below should be coming from the a Job instance.

File: plaster/run/plots/single_molecule/single_molecule_report.py~ [hunk 3]
```diff
@@ -0,0 +1,783 @@
+# Generate reports for individual traces in a run, organized into one HTML file per field/channel.
```
Comment (zack-erisyon) [inline plaster/run/plots/single_molecule/single_molecule_report.py~:1]: This needs to be removed. I assume it was added before the gitignore was setup

**Review by zack-erisyon — CHANGES_REQUESTED** [review]
I'm not sure what our plan is. If this code is going to be main-lined then it seems like we need to put some work in to making it conform to existing patterns. For example, using APIs that exist, abstract out I/O, add tests.
If we think we're throwing this in for some quick gains and plan on deprecating it, that's a different story. In which case we should add a README that states our plan.
My guess is that this code is going to have utility and isn't going  anywhere for quite some time and if that's the case then we need to either spruce it up now or schedule time for that otherwise we run the risk of leaving these dangling issues to bite us before long, esp the path issues.

**Comment by chris-erisyon** [comment]
Code updated as per our discussions:

- Lots of documentation added, including notes for future developers
- Moved the hardcoded paths to the local test code (in __main__)
- got UUID from job_manifest at the notebook level
- added _get_run_name() placeholder to get a canonical, human readable name for the runs. Will replace with final version once we have it.

-Chris

**Review by zack-erisyon — APPROVED** [review]
There's still a tilde file that should be removed

**Comment by chris-erisyon** [comment]
Removed.

-Chris

**Pull request merged**